src/data/vector/basic.lean

```diff
@@ -48,6 +48,9 @@ instance zero_subsingleton : subsingleton (vector α 0) :=
 @[simp] theorem cons_tail (a : α) : ∀ (v : vector α n), (a ::ᵥ v).tail = v
 | ⟨_, _⟩ := rfl
 
+lemma exists_eq_cons (v : vector α n.succ) : ∃ (a : α) (as : vector α n), v = a ::ᵥ as :=
+⟨v.head, v.tail, symm $ vector.cons_head_tail v⟩
```

Comment:
See above; I think this might just be a weaker version of `vector.induction_on`.